tests/e2e/framework/common.go

```diff
@@ -877,10 +877,21 @@ func (f *Framework) WaitForScanStatus(namespace, name string, targetStatus compv
 			return false, nil
 		}
 
-		if exampleComplianceScan.Status.Phase != compv1alpha1.PhaseDone && exampleComplianceScan.Status.Phase != compv1alpha1.PhasePending {
-			// check we should not have checkCount annotation
+		switch exampleComplianceScan.Status.Phase {
+		case compv1alpha1.PhaseLaunching:
+			// assert exampleComplianceScan.Annotations[compv1alpha1.ComplianceCheckCountAnnotation] is empty
 			if exampleComplianceScan.Annotations[compv1alpha1.ComplianceCheckCountAnnotation] != "" {
-				return true, fmt.Errorf("compliancescan %s has unset checkCount annotation", name)
+				return true, fmt.Errorf("compliancescan %s should not have checkCount annotation", name)
+			}
+		case compv1alpha1.PhaseRunning:
+			// assert exampleComplianceScan.Annotations[compv1alpha1.ComplianceCheckCountAnnotation] is empty
+			if exampleComplianceScan.Annotations[compv1alpha1.ComplianceCheckCountAnnotation] != "" {
+				return true, fmt.Errorf("compliancescan %s should not have checkCount annotation", name)
+			}
+		case compv1alpha1.PhaseDone:
```

Comment:
Strange - this case is causing a bunch of the tests to fail... 

```
main_test.go:1188: failed waiting for scan test-scan-with-invalid-scan-type-fails due to timeout: compliancescan test-scan-with-invalid-scan-type-fails should have checkCount annotation
```

We should certainly be setting that annotation before the scan goes into the `DONE` phase.

https://github.com/ComplianceAsCode/compliance-operator/blob/master/pkg/controller/compliancescan/compliancescan_controller.go#L666-L673

However, I wonder if it's because we don't update the annotation in the controller unless the annotation is missing?

Reply:
ahh, we are skipping the annotation when it has errors in the scan `test-scan-with-invalid-scan-type-fails`

Reply:
And that's by design, right? We don't want to set the annotation when the scan errors out?

That seems reasonable.